src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.Unix.cs

```diff
@@ -554,51 +554,25 @@ public static FileAttributes GetAttributes(string fullPath)
         }
 
         public static void SetAttributes(string fullPath, FileAttributes attributes)
-        {
-            new FileInfo(fullPath, null).Attributes = attributes;
-        }
+            => default(FileStatus).SetAttributes(fullPath, attributes, asDirectory: false);
```

Comment:
it's a little bit unusual to see syntax like this `default(SomeType).SomeMethod()` but I see where this is coming from. We might consider introducing some static helper methods in the future.